### PR TITLE
DDF-2294 Enabling Validation Filter Plugin as a feature

### PR DESCRIPTION
**Pull request by lamhuy**
#### What does this PR do?
Enabling Validation Filter Plugin as a feature
Development was done on a prior PR
https://github.com/codice/ddf/pull/3614

#### Who is reviewing it? 
@bdeining 
@coyotesqrl 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams

@bdeining
@clockard
#### How should this be tested?
Full build and Test

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2294

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by lamhuy** [comment]
build now

File: catalog/catalog-app/src/main/resources/features.xml [hunk 1]
```diff
@@ -407,6 +407,12 @@
         </bundle>
     </feature>
 
+    <feature name="catalog-core-validationfilterplugin" version="${project.version}"
```
Comment (coyotesqrl) [inline catalog/catalog-app/src/main/resources/features.xml]: ❓ Why are you adding the feature here specifically when the bundle is already started as one of the `catalog-core-plugin` here - https://github.com/codice/ddf/blob/e86ca0d388b820e5033768db13186d3312d343e3/catalog/catalog-app/src/main/resources/features.xml#L90

Comment (lamhuy) [reply catalog/catalog-app/src/main/resources/features.xml]: ah right i modified that to reference the feature now.

**Comment by lamhuy** [comment]
build now

File: catalog/catalog-app/src/main/resources/features.xml (hunk 1 above)
Comment (coyotesqrl) [reply catalog/catalog-app/src/main/resources/features.xml]: I still don't see the advantage, or why it should be started differently from the other plugins. Is the intent to make it such that downstream projects could skip the startup of `catalog-core-plugins` but start this plugin? Shouldn't we then wrap all the plugins as single-bundle features so they too could be independently managed?

Comment (lamhuy) [reply catalog/catalog-app/src/main/resources/features.xml]: yes currently GSR doesn't start the `catalog-core-plugins`. Yes perhaps we need to break all the features up but that could have downstream effect. Or perhaps we just break the `catalog-core-validationfilterplugin` to be independently managed?

Comment (coyotesqrl) [reply catalog/catalog-app/src/main/resources/features.xml]: I want to see consistency here. I'm a little surprised that your downstream project doesn't start _any_ of these other plugins; if it does, then I'd imagine it's either defined its own feature that starts the plugin bundles it needs (in which case you can just add this new plugin there) or it's defined features to contain each of the bundles individually.

Regardless, this inconsistency here in DDF is bothersome.

**Comment by lamhuy** [comment]
build now

File: catalog/catalog-app/src/main/resources/features.xml (hunk 1 above)
Comment (coyotesqrl) [reply catalog/catalog-app/src/main/resources/features.xml]: Are you going to address this? It seems premature to trigger a CI build.

Comment (lamhuy) [reply catalog/catalog-app/src/main/resources/features.xml]: you are right. i am closing this PR w/o merging. `catalog-core-plugins` feature is included by `catalog-app` which is included in `gsr-profile`, so `catalog-core-validationfilterplugin` won't be it own feature but declared as a bundle within the `catalog-core-plugins1

**Pull request closed without merging**